massdash/server/FeatureMapPlotterServer.py

```diff
@@ -0,0 +1,206 @@
+
+        return self
+
+    def generate_one_dimensional_plots(self):
+        """
+        Generates plots for each file in the transition group dictionary.
+        """
+
+
+        if self.chrom_plot_settings.display_spectrum:
+            spectra = self.featureMapCollection.to_spectra()
+            plot_settings_dict = self._get_plot_settings('m/z', 'Intensity', 'spectra')
+            self._generate_plots_helper(spectra, plot_settings_dict)
+
+        if self.chrom_plot_settings.display_chromatogram:
+            chromatograms = self.featureMapCollection.to_chromatograms()
+
+            # Perform peak picking or add features 
+            if self.peak_picking_settings.do_peak_picking == 'Feature File Boundaries':
+                tr_group_feature_data = self.chrom_loader.loadTransitionGroupFeatures(self.transition_list_ui.transition_settings.selected_peptide,
+                    self.transition_list_ui.transition_settings.selected_charge)
+
+            elif self.peak_picking_settings.do_peak_picking in ['pyPeakPickerMRM', 'MRMTransitionGroupPicker', 'Conformer']:
+                # Perform peak picking if enabled
+                peak_picker = PeakPickingServer(self.peak_picking_settings, self.chrom_plot_settings)
+                tr_group_feature_data = peak_picker.perform_peak_picking(tr_group_data=chromatograms, 
+                                                                            spec_lib=self.chrom_loader.libraryFile)
+            elif self.peak_picking_settings.do_peak_picking == 'none':
+                tr_group_feature_data = None
+                pass
+
+            else:
+                raise ValueError(f"Invalid peak picking algorithm: {self.peak_picking_settings.do_peak_picking}. Valid options are 'Feature File Boundaries', 'pyPeakPickerMRM', 'MRMTransitionGroupPicker', and 'Conformer'.")
+
+            plot_settings_dict = self._get_plot_settings('Retention Time (s)', 'Intensity', 'chromatogram')
+            self._generate_plots_helper(chromatograms, plot_settings_dict, tr_group_feature_data)
+        
+        if self.chrom_plot_settings.display_mobilogram:
+            mobilograms = self.featureMapCollection.to_mobilograms()
+            plot_settings_dict = self._get_plot_settings('Ion Mobility (1/K0)', 'Intensity', 'mobilogram')
+            self._generate_plots_helper(mobilograms, plot_settings_dict)
+
+    def generate_two_dimensional_plots(self):
+        """
+        Generates two-dimensional plots based on the targeted data.
+
+        For each file in the targeted data, if the data is not empty, it retrieves the plot settings,
+        creates an instance of InteractiveTwoDimensionPlotter, generates the two-dimensional plots,
+        and stores them in the plot_obj_dict.
+
+        Returns:
+            None
+        """
+        for file, tr_df in self.targeted_data.items():
```

Comment:
This seems to throw an error when  testing 2D plotting for the example dia data

![image](https://github.com/Roestlab/massdash/assets/32938975/b94c5bbb-3222-4ae6-a704-80004cf96a59)